### PR TITLE
ENH Highlight the tabs which have validation errors

### PR DESCRIPTION
**Pull request by GuySartorelli**
Adds the same validation tab icon and form-level message in `GridField` edit forms that you already get with validation errors in `CMSMain`.

image

## Issue
- https://github.com/silverstripe/silverstripe-admin/issues/1752

### CODE REVIEW

File: themes/cms-forms/templates/SilverStripe/Forms/CMSTabSet.ss [hunk 1]
```diff
@@ -5,15 +5,17 @@
...
-			$FieldHolder
-		<% else %>
-			<div $AttributesHTML>
-				<% loop $Fields %>
-					$FieldHolder
-				<% end_loop %>
-			</div>
-		<% end_if %>
-	<% end_loop %>
+	<div class="tab-content">
```
Comment (ChloeSartorelli) [inline themes/cms-forms/templates/SilverStripe/Forms/CMSTabSet.ss:8]: `.tab-content` has no styling implications here and isn't used for any js other than the validation tab-icon logic.
Adding the extra div similarly doesn't affect styling or js out of the box.

File: themes/cms-forms/templates/SilverStripe/Forms/CMSTabSet.ss [hunk 2]
```diff
@@ -5,15 +5,17 @@
...
+        <% loop $Tabs %>
+            <% if $Tabs %>
+                $FieldHolder
+            <% else %>
+                <div $getAttributesHTML("class") class="tab-pane $extraClass">
```
Comment (ChloeSartorelli) [inline themes/cms-forms/templates/SilverStripe/Forms/CMSTabSet.ss:13]: `.tab-pane` has no styling implications here and isn't used for any js other than the validation tab-icon logic.

File: tests/behat/src/AdminContext.php [hunk 3]
```diff
@@ -22,8 +22,8 @@ public function iShouldSeeAnInvalidTabIcon(string $not, string $tabLabel)
     {
         $id = str_replace('Main Content', 'Main', $tabLabel ?? '');
         $id = str_replace(' ', '_', $id ?? '');
-        $id = "tab-Root_$id";
-        $selector = "[aria-labelledby=$id] .font-icon-attention-1";
+        $id = "Root_$id";
+        $selector = "[aria-labelledby=tab-$id] .font-icon-attention-1, li[aria-controls=$id] .font-icon-attention-1";
```
Comment (ChloeSartorelli) [inline tests/behat/src/AdminContext.php:26]: This is needed so we can check the "invalid tab icon" in a `GridField` edit form

File: client/src/styles/legacy/_style.scss [hunk 4]
```diff
@@ -688,7 +688,10 @@ html, body {
 
 .validationerror .panel--padded .ui-tabs-nav {
   margin-top: -$panel-padding-y;
+}
 
+.validationerror .panel--padded .ui-tabs-nav,
+.validationerror .toolbar--north .ui-tabs-nav {
   .tab-attention {
     color: $state-draft;
     position: absolute;
```
Comment (ChloeSartorelli) [inline client/src/styles/legacy/_style.scss:697]: `.panel--padded` isn't present in `GridField` edit forms, and would be at the wrong level even if it was present. This is needed mostly to make sure the tab icon has the correct colour.

File: client/src/legacy/LeftAndMain.EditForm.js [hunk 5]
```diff
@@ -121,7 +121,7 @@ $.entwine('ss', function($){
           'There are validation errors on this page, please fix them before saving or publishing.'
         );
 
-        const $editFormErrorBanner = $("#Form_EditForm_error");
+        const $editFormErrorBanner = $("#Form_EditForm_error, #Form_ItemEditForm_error");
```
Comment (ChloeSartorelli) [inline client/src/legacy/LeftAndMain.EditForm.js:124]: Would be better to not rely on IDs but I'm going for the minimally invasive solution here, and `GridField`'s edit form always has the same ID.

**Review by emteknetnz — CHANGES_REQUESTED** [review]
This works, though has the same bug reported on https://github.com/silverstripe/silverstripe-cms/issues/2944 where if you have say an invalid Email address, the tab it failed on isn't highlighted and there isn't a form level message.  RequiredFields on the other hand does work correctly, also just like in the linked issue

**Comment by GuySartorelli** [comment]
That bug will be fixed by the pr that fixes that bug.

**Comment by emteknetnz** [comment]
I've merged the other one, could you give this one a rebase and just confirm on your local it works as expected?

**Comment by GuySartorelli** [comment]
Rebased. Checked locally that it works as expected.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**